src/libstd/macros.rs

```diff
@@ -117,7 +117,33 @@ macro_rules! println {
 }
 
 /// Helper macro for unwrapping `Result` values while returning early with an
-/// error if the value of the expression is `Err`.
+/// error if the value of the expression is `Err`. Can only be used in
+/// functions that return `Result` because of the early return of `Err` that
+/// it provides.
+///
+/// # Examples
+///
+/// ```
+/// use std::io;
+/// use std::fs::File;
+///
+/// fn write_to_file_using_try() -> Result<(), io::Error> {
+///     let mut file = try!(File::create("my_best_friends.txt"));
+///     try!(file.write_line("This is a list of my best friends."));
+///     println!("I wrote to the file");
+///     Ok()
+/// }
+/// // This is equivalent to:
+/// fn write_to_file_using_match() -> Result<(), io::Error> {
+///     let mut file = try!(File::create("my_best_friends.txt"));
+///     match file.write_line("This is a list of my best friends.") {
+///       Ok(_) => (),
+///       Err(e) => return Err(e),
```

Comment:
Could this be sure to use 4-space tabs instead of 2-space tabs?

Also, I think the `write_line` function has since been removed (old I/O?), but perhaps `write_all` with a byte-string literal could be used? I think this example may also need to import `std::io::prelude::*` to get the `Write` trait.